test-framework/common/src/main/java/io/quarkus/test/common/TestResourceManager.java

```diff
@@ -83,6 +84,11 @@ public void stop() {
                 throw new RuntimeException("Unable to stop Quarkus test resource " + testResource, e);
             }
         }
+        ConfigProviderResolver cpr = ConfigProviderResolver.instance();
+        try {
+            cpr.releaseConfig(cpr.getConfig());
```

Comment:
I'm surprised it's the responsibility of this class to do it?

Reply:
Yeah I wasn't sure how this architecture was intended to work.  I feel like we should have a consistent way to set up the configuration but we do not yet.  So again I opted to preserve the existing behavior as much as possible.

Reply:
#5702